Create data model and migration

kadmit opened the pull request:
- Created database migration.
- Created data model with Eloquent ORM.
- Created tests for testing data model.

Commit:
chore: create data model for classifier value

Commit:
Merge branch 'feat/setup-dockerized-classifier-service' into feat/create-data-models

Commit:
refactor: Add default value for classifier value id && changed type of `type` column to enum.

Commit:
Merge branch 'main' into feat/create-data-models

DanBondarenko commented on application/app/Traits/Uuid.php:
Laravel has a built-in trait for the same purpose: `Illuminate\Database\Eloquent\Concerns\HasUuids` ([see docs](https://laravel.com/docs/10.x/eloquent#uuid-and-ulid-keys)). Probably better to use the built-in one and remove this one (the "less code the better" approach).

DanBondarenko commented on application/database/migrations/2023_03_30_143724_create_classifier_value_table.php:
```diff
@@ -0,0 +1,34 @@
+    {
+        Schema::create('classifier_values', function (Blueprint $table) {
+            $table->uuid('id')->primary()->default(DB::raw('gen_random_uuid()'));
+            $table->timestamps();
```
Seems like the `deleted_at` column has timezone information, but `created_at` and `updated_at` columns do not. Better to go for a consistent approach. Laravel’s documentation [recommends always including](https://laravel.com/docs/10.x/eloquent-mutators#date-casting-and-timezones) timezone information, so I would suggest: 

```suggestion
            $table->timestampsTz();
```

Quote from docs:
> By default, the date and datetime casts will serialize dates to a UTC ISO-8601 date string (1986-05-28T21:05:54.000000Z), regardless of the timezone specified in your application's timezone configuration option. You are strongly encouraged to always use this serialization format, as well as to store your application's dates in the UTC timezone by not changing your application's timezone configuration option from its default UTC value. Consistently using the UTC timezone throughout your application will provide the maximum level of interoperability with other date manipulation libraries written in PHP and JavaScript.

DanBondarenko commented on application/database/migrations/2023_03_30_143724_create_classifier_value_table.php:
```diff
@@ -0,0 +1,34 @@
+     */
+    public function down(): void
+    {
+        Schema::dropIfExists('classifier_value');
```
Looks like a typo. Consequently, `php artisan migrate:refresh` currently fails.

```suggestion
        Schema::dropIfExists('classifier_values');
```

DanBondarenko commented on application/tests/Unit/.gitkeep:
Hmm, just to be sure, what will be the purpose of this file? Do we need it in the repository?

thenouan replied:
Yes, this serves as a placeholder to keep the folder in version control

Commit:
refactor: Renamed trait && fixed migration typo

DanBondarenko reviewed (changes requested):
Hi @kadmit, I looked over the pull request and found a few opportunities for improvements. I left specific comments on files and I’ll also copy some more general comments below:

### Running

I tried running the application using the Docker configuration in the repository. Unfortunately it seems that there might be a mistake in the Docker configuration since running `docker compose -f docker-compose.development.yml up` results in the following error.

```
[+] Running 1/0
 ⠿ Container tv-classifier-app-1  Created
Attaching to tv-classifier-app-1
tv-classifier-app-1  | sh: composer: not found
tv-classifier-app-1 exited with code 127`
```

I got it running with Sail instead (`php artisan sail:install`, `cp .env.example .env`, `vendor/bin/sail up` etc).

### Styling

I noticed that some files do not pass [the Pint style check](https://laravel.com/docs/10.x/pint). Seems like following the Pint styleguide might be a decent approach (haven’t discussed this with Markus yet though). It’s really easy to use Pint to fix files, just run `vendor/bin/pint`.

```
vendor/bin/pint --test

    FAIL   ............................................................. 47 files, 4 style issues
  ⨯ app/Models/ClassifierValue.php       class_attributes_separation, trailing_comma_in_multiline
  ⨯ database/factories/ClassifierValueFactory.php braces, trailing_comma_in_multiline, blank_lin…
  ⨯ routes/api.php                                        no_unused_imports, no_extra_blank_lines
  ⨯ routes/web.php                                        no_unused_imports, no_extra_blank_lines
```

thenouan commented:
> Hi @kadmit, I looked over the pull request and found a few opportunities for improvements. I left specific comments on files and I’ll also copy some more general comments below:
> 
> ### Running
> I tried running the application using the Docker configuration in the repository. Unfortunately it seems that there might be a mistake in the Docker configuration since running `docker compose -f docker-compose.development.yml up` results in the following error.
> 
> ```
> [+] Running 1/0
>  ⠿ Container tv-classifier-app-1  Created
> Attaching to tv-classifier-app-1
> tv-classifier-app-1  | sh: composer: not found
> tv-classifier-app-1 exited with code 127`
> ```
> 
> I got it running with Sail instead (`php artisan sail:install`, `cp .env.example .env`, `vendor/bin/sail up` etc).
> 

@DanBondarenko I would keep away from Laravel sail

Commit:
refactor:  Pint style fixes

Commit:
refactor:  Remove custom trait for uuids && add usage of  laravel HasUUids trait.

Commit:
refactor:  Add timezone for `created_at` and `updated_at` columns.

Commit:
fix: Remove running of composer as entrypoint.

kadmit commented:
@DanBondarenko thank you for the review!

Regarding this part:
> I tried running the application using the Docker configuration in the repository. Unfortunately it seems that there might be a mistake in the Docker configuration since running docker compose -f docker-compose.development.yml up results in the following error.

Looks like you used the wrong command to run docker-compose. The right one is ` docker compose -f docker-compose.yml -f docker-compose.development.yml up`. The main purpose here is to have the possibility to run locally the app with the production and development configurations. When the additional docker-compose configuration file is passed it adds/overrides the main configuration file parameters. 

To make development more convenient I added `.env` file that defines docker-compose config file and convert this command `docker compose -f docker-compose.yml -f docker-compose.development.yml up` into `docker compose up`.

To fix the issue I removed the running of the composer as the entry point.

DanBondarenko commented:
> Looks like you used the wrong command to run docker-compose. The right one is ` docker compose -f docker-compose.yml -f docker-compose.development.yml up`. The main purpose here is to have the possibility to run locally the app with the production and development configurations. When the additional docker-compose configuration file is passed it adds/overrides the main configuration file parameters.

@kadmit A-ha, I see, my bad. Thank you for the clarification. :) 

Anyways, thank you for the changes and looks good to me – approved!
